bittide-extra/src/Bittide/Extra/Maybe.hs

```diff
@@ -0,0 +1,47 @@
+-- SPDX-FileCopyrightText: 2023 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+module Bittide.Extra.Maybe where
+
+import Clash.Prelude
+
+import Data.Maybe
+
+{- $setup
+>>> import Clash.Prelude
+-}
+
+-- | Extract `Just` from a `Vec` of `Maybe`s. Prefer left-most `Just`. If no `Just` is
+-- found, use a default value.
+--
+-- >>> fromMaybesL 0 (Just 1 :> Just 2 :> Nothing :> Nil)
+-- 1
+-- >>> fromMaybesL 0 (Nothing :> Nothing :> Nothing :> Nil)
+-- 0
+--
+fromMaybesL :: a -> Vec n (Maybe a) -> a
+fromMaybesL a = fromMaybe a . fold (<|>) . (Nothing :>)
+
+-- | Extract `Just` from a `Vec` of `Maybe`s. Prefer right-most `Just`. If no `Just` is
+-- found, use a default value.
+--
+-- >>> fromMaybesR 0 (Just 1 :> Just 2 :> Nothing :> Nil)
+-- 2
+-- >>> fromMaybesR 0 (Nothing :> Nothing :> Nothing :> Nil)
+-- 0
+--
+fromMaybesR :: a -> Vec n (Maybe a) -> a
+fromMaybesR a = fromMaybe a . fold (flip (<|>)) . (Nothing :>)
+
+-- | Returns 'Just a' when the boolean is 'True', or 'Nothing' when 'False'.
```

Comment:
Since `bittide` depends on `clash-lib`, you could also just import `clash-lib:Clash.Netlist.Util.orNothing`. But maybe you're setting the stage for adding this module to `clash-prelude` and that's why you are copying the existing function from `clash-lib` to here, to ultimately copy to `clash-prelude`?